pkg/option/config.go

```diff
@@ -1356,9 +1356,16 @@ func LogRegisteredOptions(vp *viper.Viper, entry *logrus.Entry) {
 	keys := vp.AllKeys()
 	sort.Strings(keys)
 	for _, k := range keys {
-		v := vp.GetStringSlice(k)
-		if len(v) > 0 {
-			entry.Infof("  --%s='%s'", k, strings.Join(v, ","))
+		ss := vp.GetStringSlice(k)
```

Comment:
Should we move this commit into a separate PR so that it can be backported to all branches?

Reply:
It actually seems that `flags.StringToString` should not be used (https://github.com/cilium/cilium/blob/main/contrib/scripts/check-viper.sh#L20-L23). Let me try to figure out how to make `option.NewNamedMapOptions` work for  hive flags. In that way, this commit should no longer be necessary.

Reply:
I've opened https://github.com/cilium/cilium/pull/25643 which includes the logic to correctly parse StringMap flags through the hive machinery, and the commit to fix the logging of that type of flags. Backporting is not required, since StringMap flags are not currently used (an intermediate wrapper is used to support both json and KV encoding for legacy flags).